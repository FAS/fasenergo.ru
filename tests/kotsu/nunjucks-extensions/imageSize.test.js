/* eslint-env jest */

import { renderString } from '../../utils/nunjucks'

const render = (template, context = mockContext , parse) => renderString(template, context, parse)
const mockContext = {
  PATH: {
    images: 'assets/images',
    build: {
      root: 'build'
    }
  },
  SITE: {
    __images: [
      { 'name': 'build/assets/images/test.jpg', 'width': 100, 'height': 150 },
      { 'name': 'build/assets/images/test2.png', 'width': 120, 'height': 170 }
    ]
  }
}

describe('Nunjucks `imageSize()`', () => {
  it('should get image size', () => {
    expect(render(`
      {% set image = imageSize('/' + PATH.images + '/test.jpg') %}
      <img src='{{ image.src }}' width='{{ image.width }}' height='{{ image.height }}'>
    `)).toMatchSnapshot()
  })

  it('should error on unknown image', () => {
    expect(() => render(`{% set size = imageSize('/nope.jpg') %}`)).toThrowErrorMatchingSnapshot()
  })
})
