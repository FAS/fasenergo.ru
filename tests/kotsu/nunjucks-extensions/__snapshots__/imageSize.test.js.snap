// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nunjucks \`imageSize()\` should error on unknown image 1`] = `
"(unknown path)
  Error: [imageSize] image \`/nope.jpg\` did not match any image data"
`;

exports[`Nunjucks \`imageSize()\` should get image size 1`] = `
"
      
      <img src='/assets/images/test.jpg' width='100' height='150'>
    "
`;

exports[`Nunjucks \`imageSize()\` should print srcset 1`] = `
"
      
      <img src='/assets/images/test.jpg'
        srcset=''
      >

      
      <img src='/assets/images/set.jpg'
        srcset='/assets/images/set.jpg 2000w, /assets/images/set@500.jpg 500w, /assets/images/set@1000.jpg 1000w'
      >
    "
`;
